docs-site/docs/contribute-code.md

```diff
@@ -19,16 +19,26 @@ A simple way to enforce the Google coding styles is to use the linting and forma
 Once the extensions are installed, edit the `settings.json` file.
 
 ```json 
-"[cpp]": {
-    "editor.defaultFormatter": "xaver.clang-format"
-},
-"cpplint.cpplintPath": "path/to/cpplint",
-
-"editor.formatOnSave": true,
-"python.formatting.provider": "yapf",
-"python.linting.enabled": true,
-"python.linting.lintOnSave": true,
+{
+   "[cpp]": {
+      "editor.defaultFormatter": "xaver.clang-format"
+   },
+   "cpplint.cpplintPath": "path/to/cpplint",
+
+   "editor.formatOnSave": true,
+   "python.formatting.provider": "yapf",
+   "python.linting.enabled": true,
+   "python.linting.lintOnSave": true,
+   "clang-format.language.cpp.style": "google",
+   "python.formatting.yapfArgs" : ["--style" , "{based_on_style: google}" ]
+}
 ```
+
+Depending on your platform, the user settings file is located here:
+1. Windows %APPDATA%\Code\User\settings.json
+2. macOS $HOME/Library/Application Support/Code/User/settings.json
+3. Linux $HOME/.config/Code/User/settings.json
+
 Configurations are specified in corresponding config files. And these tools would look up for configuration files in the root of the project automatically, e.g. `.pylintrc`.
```

Comment:
do the settings in line 22-34 lead to the same format result as the configuration file?

Reply:
Yes, this is the intention of this PR. It will solve the problem when the project root .clang-format is not detected in some environment (e.g. sftp of WINSCP)
the key is the line 22 and 23 that set the format to google
```
"clang-format.language.cpp.style": "google",
"python.formatting.yapfArgs" : ["--style" , "{based_on_style: google}" ]
```
I have checked the .clang-format just have two line that turns the style to google
https://github.com/apache/singa/blob/dev/.clang-format

However, I cannot guarantee in all case the formatting is the same by the two approaches.